pkg/util/util.go

```diff
@@ -626,3 +628,20 @@ func DeletePath(path string) error {
 	}
 	return nil
 }
+
+// HTTPGetRequest uses url to get file contents
+func HTTPGetRequest(url string) ([]byte, error) {
```

Comment:
We should implement tests here, there is code for testing this located here: https://itnext.io/http-request-timeouts-in-go-for-beginners-fe6445137c90

Reply:
Thanks for the providing the testing resource! This test is in my test automation TODO list. The main issue can be divided into two PRs: `odo catalog list components` and `odo create <component type>`, I do have a followup PR for covering all test automation because codes may change before that PR, that's why I didn't include the tests in this PR and we can use `Experimental` flag to disable the feature. (No regression).

My plan is:
1. I merge `odo catalog list components` PR
2. You merge the format PR with corresponding tests you want to add
3. I merge the `odo create <component type>` PR with corresponding tests I want to add